src/main/java/com/nextcloud/common/NextcloudClient.kt

```diff
@@ -107,9 +107,9 @@ class NextcloudClient(var baseUri: Uri, val context: Context) : OkHttpClient() {
                     destination = redirectedDestination
 
                     if (getRequestHeader("Destination").isNullOrEmpty()) {
-                        method.requestHeaders.put("destination", destination)
+                        method.setRequestHeaders("destination" to destination)
```

Comment:
This changes the behavior since the original code just replaces the destination value, while this call leads to a `requestHeaders.clear()` clearing the whole header collection.

Reply:
So `addRequestHeader(...)` should lead to the same behavior afaik.

Reply:
Indeed, it should be added, but not remove all other headers.